File: 465/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_VAL 0x7fffffff

int main(int argc, char* argv[])
{
   double a, b;
   char op;
   char number[2][1000];

   while (scanf("%s %c %s", number[0], &op, number[1]) == 3) {
      printf("%s %c %s\n", number[0], op, number[1]);
      a = atof(number[0]);
      b = atof(number[1]);
      if (a > MAX_VAL) printf("first number too big\n");
      if (b > MAX_VAL) printf("second number too big\n");
      if (op == '+' && (a + b) > MAX_VAL)
         printf("result too big\n");
      if (op == '*' && (a * b) > MAX_VAL)
         printf("result too big\n");
   }

   return 0;
}

File: 465/Main.java
import java.io.*;
import java.util.*;

public class Main {
   public static void main(String args[]) throws IOException {
      String line;
      StringTokenizer tokenizer;
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      while ((line = in.readLine()) != null) {
         System.out.println(line);
         tokenizer = new StringTokenizer(line);
         double a = Double.parseDouble(tokenizer.nextToken());
         char operator = tokenizer.nextToken().charAt(0);
         double b = Double.parseDouble(tokenizer.nextToken());
         if (a > Integer.MAX_VALUE) System.out.println("first number too big");
         if (b > Integer.MAX_VALUE) System.out.println("second number too big");
         if (operator == '+' && (a + b) > Integer.MAX_VALUE)
            System.out.println("result too big");
         if (operator == '*' && (a * b) > Integer.MAX_VALUE)
            System.out.println("result too big");
      }
   }
}
